packages/core/src/data_sources/index.ts

```diff
@@ -148,4 +149,59 @@ export default class DataSourceManager extends ItemManagerModule<ModuleConfig, D
 
     return result;
   }
+
+  /**
+   * Store data sources to a JSON object.
+   * @returns {Object} Stored data sources.
+   */
+  store() {
+    const data: ObjectAny = {};
+    this.all.forEach((dataSource) => {
+      const shouldStoreInProject = dataSource.get('shouldStoreInProject');
+      if (shouldStoreInProject) {
+        data[dataSource.id] = {
+          id: dataSource.id,
+          name: dataSource.get('name' as any),
+          records: dataSource.records.toJSON(),
+          shouldStoreInProject,
+        };
+      }
+    });
+
+    return { [this.storageKey]: data };
+  }
+
+  clear(): this {
+    // Clearing data sources are a no-op as to preserve data sources.
+    // This is because data sources are optionally stored in the project data.
+    // and could be defined prior to loading the project data.
+    return this;
+  }
+
+  /**
+   * Load data sources from a JSON object.
+   * @param {Object} data The data object containing data sources.
+   * @returns {Object} Loaded data sources.
+   */
+  load(data: any) {
+    const storedDataSources: Record<string, DataSourceProps> = data[this.storageKey] || {};
+    const memoryDataSources = this.em.DataSources.getAllMap();
+
+    if (!Object.keys(storedDataSources).length) {
+      return {
+        ...memoryDataSources,
+      };
+    } else {
+      this.clear();
+
+      Object.values(storedDataSources).forEach((ds) => {
+        this.add(ds, { silent: true });
+      });
+
+      return {
+        ...storedDataSources,
+        ...memoryDataSources,
+      };
+    }
+  }
```

Comment:
I'd expect this to be similar to other storable modules, eg.
```js
load(data: ProjectData) {
  return this.loadProjectData(data);
}
```
any reason of this custom implementation?

Reply:
My thinking is that Data Sources could be fetched and defined before loading the project JSON.
Consider the example of loading a large JSON structure from an API. You may want to fetch and load the data into memory and into a data source first, then load the project JSON on top of it. Without this order of operations, you could experience flickering in the UI as it displays default values.

Additionally, you may have a combination of data sources stored in the project JSON, such as global styles, along with larger ones stored dynamically. This approach ensures that they are loaded correctly and not overridden depending on the order of operations.

Reply:
Yeah, I guess there will be different logic and approaches to managing dynamic data sources but I'd leave that to the consumer (eg. hide the canvas until all the dynamic data sources are loaded).

All current storable modules should work in the same way. When you do `editor.loadProjectData(json)`, this is the request to load a completely new project, so you expect to `clear` all the current data sources and load those in the json (eg. if you have `dataSources: []`, that means the project doesn't contain any).
For sure someone will think one day to load partial data sources from another project but that logic should be handled on the consumer side (eg. the consumer loads the JSON of another project and decides what to add to the current one)